### ISSUE TITLE
MVP

### ISSUE THREAD

**Issue by hetec** [socobo/SocoboPlusPlus#113]
We should define the feature set for our mvp I think! In this sense I would suggest to stop the backend development at some point and start implementing the frontend to get some minimal product.

On basis of this minimal product we can than go ahead. An it's maybe a lot easer to define new features like the rating, sharing, grouping ... if we have a clue how our product will work in action and how it looks.

What do you think?

**Comment by flashback2k14** [comment]
Yes, I'm with you. After implementing issue #95 and #108 we should feature freeze the backend and start frontend development. But we should start the discuss about our layout, color schema, fonts or the CSS framework.

**Comment by hetec** [comment]
Yes your are right we should discuss this beforehand

**Comment by flashback2k14** [comment]
@hetec i have started creating some mockup idea's [here](https://drive.google.com/file/d/1XpZqpo7KFJ2fB59LnffKiUBl_w-68407/view?usp=sharing). Maybe you could add yours too!?

**Comment by patrick-hebner** [comment]
![recipe_overview](https://user-images.githubusercontent.com/6704069/35481865-7dddc416-042c-11e8-98e0-8cf311217ec0.png)

**Comment by patrick-hebner** [comment]
I would prefer a card like layout like this one, because it enables the user to get a quick overview over all recipes. I would also enable direct access to filters, because I think this is one of the most important feature for searching a recipe page

**Comment by flashback2k14** [comment]
what tool did you use?

yes, we could use the card grid view. looks nice. what did you think about my idea about the recipe carousel? 

I would add the filter under or above the search bar. and the categories should be added into a combobox. something like this:

![bildschirmfoto 2018-01-28 um 15 56 39](https://user-images.githubusercontent.com/6609359/35483561-e2e61072-0443-11e8-90bd-883245a849fd.png)

**Comment by patrick-hebner** [comment]
For this I used the trial of Sketch, but Figma is also a very nice free tool for this.

**Comment by patrick-hebner** [comment]
To your idea:

- technically sliders are a nice solution but they are often bad for ux, because
- they only provide a only a small amount of the recipes to the user
- To scroll horizontally or use the arrows is a very confusing and unnatural way to navigate through the recipes (compared to simply scrolling)
- Only one recipe is fully visible, the remaining visible recipes have always hidden information, so the user needs to do an action (navigate) to see all the content
- This side also provides some reasons why we should avoid sliders https://getpocket.com/a/read/547046461

**Comment by patrick-hebner** [comment]
To the filtering solutions:

- I also thought about the solution to put filters at the top. Some reasons I decided against that:
CONTRA
- The space is very limited => If we have for example more than three levels or some new filters we will get into some problems
- Its hard to make this good at mobile, the sidebar can be simply hidden behind a toggle btn
- I wanted to use labels since it's a good practice an makes the different filter sections much more understandable to the user
PRO
- Its looks better
- We could place more recipes above the fold

Why do you want to use the multiselect? I saw some disadvantages:
- The user needs a click to reach the options
- The user doesn't see what categories are available at a glance
- I multiple categories are selected, how do you show them to the user if the select is closed?
Maybe I'm wrong?! Could you explain to me why the select is better?

And please don't hide the filters: the user will have a hard time to find them and always needs a click to use them => at least as long we have enough space.

I hope this makes clear why I designed the mockup like I did. After I know your reasons we will be able to combine the best of both solutions. What do you think. I think we should always favor UX for the user over the beauty of the UI. So first optimize UX then UI

**Comment by flashback2k14** [comment]
I can't read your article. I didn't have a pocket account. You know I have my own link collector. :-) But I get your points and it's okay. I don't need a carousel. :-) 

Okay, I followed the common search and filter practise. Very page I know at the time hide the filter and open a dialog or something else. 

Why put the categories into a select. Because if we list all categories this could be very long and other filters maybe not visible. 

Maybe we could show the categories as clickable batches!? 

How would your filter look on mobile devices?

**Comment by patrick-hebner** [comment]
https://yoast.com/opinion-on-sliders/

**Comment by patrick-hebner** [comment]
Hi, I thought we only have a limited set of categories (maybe max. 8 or so). But if there will be more you are right, the space could be a problem. But keep in mind, if the select grows (15 and more) and the user needs to scroll thats very bad too. And we should also make visible what the user selected (because thats a problem with multi select fields) But I think there are solution too, like selects that show the selected items as tags or so.

**Comment by patrick-hebner** [comment]
Nice article about UX and why u shouln't hide filters https://www.newfangled.com/using-filters-for-good-ux/

**Comment by patrick-hebner** [comment]
Maybe we also don't need so much filters and can reduce them. Then we could place them in a horizontal bar at the top. I will think about this.

**Comment by flashback2k14** [comment]
What did you think about this filter. that's are clickable batches or chips and instead of the hearts we could use chef hat!?

![bildschirmfoto 2018-01-29 um 22 09 43](https://user-images.githubusercontent.com/6609359/35534572-40efcc46-0541-11e8-968d-9760abc93d6c.png)

**Comment by patrick-hebner** [comment]
Yes I like this. I try to incorporate this in my mockup. Did you create this or is that some kind of component, plugin or so?

**Comment by flashback2k14** [comment]
I have you invite to our figma team today. Yes, It's a component.

**Comment by patrick-hebner** [comment]
![recipe_overview](https://user-images.githubusercontent.com/6704069/35591036-ea173cb0-0608-11e8-8179-0f138441ffe6.png)

**Comment by patrick-hebner** [comment]
- Big heart: Short filter for all your favorites
- Favorites are marked by a little white heart => The button is in this case a empty heart to remove a recipe from the favorites (else no white heart and a button with full heart)
- Maybe a general sorting possibility is better as a single rating filter?

I try to import the sketch file to figma

**Comment by flashback2k14** [comment]
the import should work i think. 

what if we use use chef hat instead of a heart?

**Comment by patrick-hebner** [comment]
Yes we can do this, I just had no one available

**Comment by patrick-hebner** [comment]
I think with your filter idea a horizontal filter bar above the recipes would also be possible.

**Comment by flashback2k14** [comment]
[maybe this here?] (https://www.flaticon.com/free-icon/chef_129226#term=chef hat&page=1&position=5)

**Comment by patrick-hebner** [comment]
Yes, we can start with that, looks good. Maybe we should anyway encapsulate this into a component, so we can easily change the icon if we want.

**Comment by patrick-hebner** [comment]
Can you give me a link to the component for the clickable batches?

**Comment by flashback2k14** [comment]
It's inside Figma. it's called Recipe-Filter. You could copy it and paste it into your component / file.
![screen](https://user-images.githubusercontent.com/6609359/35608901-7a559522-065b-11e8-9f42-305250df8f16.PNG)

**Comment by patrick-hebner** [comment]
Ahhhh, I actually meant if there is a ready to use angular component for this type of filter. Little misunderstanding

**Comment by flashback2k14** [comment]
Angular material has a chip component - https://material.angular.io/components/chips/overview
